protocol/0053-PERP-product_builtin_perpetual_future.md

```diff
@@ -4,12 +4,16 @@ This built-in product provides perpetual futures that are cash-settled, i.e. the
 
 [Background reading](https://www.paradigm.xyz/2021/05/everlasting-options/#Perpetual_Futures)
 
-Perpetual futures are a simple "delta one" product.
+Perpetual futures are a simple "delta one" product. Mark-to-market settlement occurs with a predefined frequency as per [0003-MTMK-mark_to_market_settlement](0003-MTMK-mark_to_market_settlement.md).  Additionally, a settlement using external data is carried out whenever `settlement_cue` is triggered AND the `settlement_data` is received. A number of protective measures can be specified for a market to deal with data scarcity in a predefined way.
 
 ## 1. Product parameters
 
-1. `settlement_data (Data Source: number)`: this data is used by the product to calculate periodic settlement cashflows. The receipt of this data triggers this calculation and the transfers between parties to "true up" to the external reference price.
 1. `settlement_asset (Settlement Asset)`: this is used to specify the single asset that an instrument using this product settles in.
+1. `settlement_cue (Data Source)`: this data is used to indicate that next periodic settlement should happen imminently.
+1. `settlement_data (Data Source: number)`: this data is used by the product to calculate periodic settlement cashflows. The receipt of this data triggers this calculation and the transfers between parties to "true up" to the external reference price.
+1. `settlement_cue_auction_duration`: a time interval which specifies the duration of an auction started once settlement cue is received. The auction ends when the specified time elapses or when the settlement data is received. A value of `0s` indicates no auction.
+1. `max_settlement_gap`: a time interval which specifies the amount of time without periodic settlement after which the market will go into protective auction and remain in that mode until settlement data is received.
+1. `settlement_price_monitoring`: a boolean flag indicating if periodic settlement price should go through the [price monitoring](0032-PRIM-price_monitoring.md) logic. If set to `true` any valid `settlement_data` ingested by the market will go through the price monitoring engine and contribute to its price history as well as trigger a price monitoring auction if it falls outside the current valid price bounds.
```

Comment:
So if the settlement value comes outside bounds then what happens to mark-to-market? Let's say bounds are [80,120] I get settlement data with 125 and the `settlement_price_monitoring = true`. Now we go into price monitoring auction if:
1) price monitoring resolves at say 124 or 126 - this value is used for mark to market. 
2) price monitoring resolves at say 100 - this value is used for mark to market (I guess). 
So then the settlement data ends up effectively not contributing to settlement, so the time we've not settled goes past `max_settlement_gap` we go into an auction, correct?

Reply:
That depends on how we resolve the TODO at line 55. 

I think what we'd like to end up with is:
1. if settlement data is very different from previous settlement prices or market prices we want to discard it (I think it makes sense to not use at all if we don't use it immediately (otherwise we're deliberately making the data stale).
1. if it's persistently different (a few consecutive ones are similar) we want to eventually accept it.
1. if it goes all over the place then some form of manual intervention wouldn't be terrible.

I think price monitoring bounds + max_settlement_gap that's not larger than say 2-3 settlement anniversaries gets us mostly there. 

An open question is if we want to use the discarded data in history, but it'd just ignore it not to create a special case in the engine and assume that if that's indeed a sensible price point then auction will resolve not far from it).

If oracle supplies consecutive datapoints that are all over the place, then it probably makes sense to either keep market in auction for longer or change the oracle (including a change to the internal oracle that just supplies a value that's been agreed on via consensus).

Reply:
It still feels somewhat wrong for the market to have the ability to largely ignore a settlement price given that those settlements are the only things giving the trading value 😄. But I guess not the end of the world as long as it's settling frequently